app/models/profile.rb

```diff
@@ -3,12 +3,47 @@ class Profile < ApplicationRecord
 
   validates :user_id, uniqueness: true
 
-  # This method generates typed accessors for all profile fields
-  def self.refresh_store_accessors!
+  USER_MIXIN = Module.new
+
+  # NOTE: @citizen428 This is a temporary mapping so we don't break DEV during
+  # profile migration/generalization work.
+  MAPPED_ATTRIBUTES = {
+    brand_color1: :bg_color_hex,
+    brand_color2: :text_color_hex,
+    display_email_on_profile: :email_public,
+    display_looking_for_work_on_profile: :looking_for_work_publicly,
+    git_lab_url: :gitlab_url,
+    linked_in_url: :linkedin_url,
+    recruiters_can_contact_me_about_job_opportunities: :contact_consent,
+    stack_overflow_url: :stackoverflow_url
+  }.with_indifferent_access.freeze
+
+  # Generates typed accessors for all currently defined profile fields.
+  def self.refresh_attributes!
     ProfileField.find_each do |field|
-      store_attribute :data, field.attribute_name, field.type
+      attribute_name = field.attribute_name
+      store_attribute :data, attribute_name, field.type
+
+      getter = MAPPED_ATTRIBUTES.fetch(attribute_name, attribute_name).to_s
+      USER_MIXIN.instance_eval do
+        define_method(getter) do
+          if profile.respond_to?(attribute_name)
+            profile.public_send(attribute_name)
+          else
+            self[getter]
+          end
+        end
+
+        # Make mapped attributes available under both names
+        alias_method(attribute_name, getter) if attribute_name != getter
+      end
     end
   end
 
-  refresh_store_accessors!
+  refresh_attributes!
+
+  # Returns an array of all currently defined `store_attribute`s on `data`.
+  def self.attributes
```

Comment:
I'm using this now outside the class instead of directly accessing `Profile.stored_attributes[:data]`. The old way broke encapsulation as it relied on an implementation detail of `Profile`.

Reply:
This would be overwriting ActiveRecord's `.attributes` right? Is this intended?

Reply:
AR's `attributes` is an instance method, this is a class method. The name was chosen intentionally, to reflect the relationship (i.e. a user's instance variable attributes are the attributes of their profile).

```ruby
User.attributes
# NoMethodError: undefined method `attributes' for #<Class:0x0000000114140a90>
User.first.attributes
#  User Load (2.3ms)  SELECT "users".* FROM "users" ORDER BY "users"."id" ASC LIMIT $1  [["LIMIT", 1]]
# {
#                          "articles_count" => 5,
#                           "available_for" => nil,
```

Reply:
Ahhh duh haha, thanks for clarifying.